frontend/catalyst/pennylane_extensions.py

```diff
@@ -126,11 +127,23 @@ def __call__(self, *args, **kwargs):
                 name = self.device.short_name
             else:
                 name = self.device.name
-            if name not in QFunc.RUNTIME_DEVICES:
+
+            is_known_device = name in QFunc.RUNTIME_DEVICES
+            implements_c_interface = hasattr(self.device, "get_c_interface")
+            is_valid_device = is_known_device or implements_c_interface
+            if not is_valid_device:
                 raise CompileError(
                     f"The {name} device is not " "supported for compilation at the moment."
                 )
 
+            backend_path_or_name = name
```

Comment:
What's the use case of the device `name` after fully implement this support in the runtime? We plan to expand this work for all of our current backend devices and then they all should be accessible from the frontend by their path to associated shared libraries.  :thinking:

Reply:
See here: https://github.com/PennyLaneAI/catalyst/pull/327/files#diff-8754bab66a7845aa8e14eb1ab898192548b9ab878d595f57b9c6639930e2c6bfL145

Reply:
I see! It may be good to add a comment here so that we revisit this when we fully adapt this support in the runtime.

Reply:
See here: https://github.com/PennyLaneAI/catalyst/pull/327/commits/6affae2e1e2a1b1bd842c4918e6c1425da1a74d0